src/rpc/format/cbor/convert.rs

```diff
@@ -263,9 +288,29 @@ impl TryFrom<Value> for Cbor {
 				Ok(Cbor(Data::Tag(TAG_DURATION, Box::new(Data::Text(v.to_raw())))))
 			}
 			Value::Datetime(v) => {
-				Ok(Cbor(Data::Tag(TAG_DATETIME, Box::new(Data::Text(v.to_raw())))))
+				if SERIALIZATION_V2 {
+					let timestamp = v.timestamp();
+					let nanos = v.timestamp_subsec_nanos();
+
+					const NANOS_PER_SEC: f64 = 1_000_000_000.0;
+
+					let data = match nanos {
+						0 => Box::new(Data::Integer(timestamp.into())),
+						_ => Box::new(Data::Float(timestamp as f64 + nanos as f64 / NANOS_PER_SEC)),
+					};
+
+					Ok(Cbor(Data::Tag(TAG_DATETIME, data)))
+				} else {
+					Ok(Cbor(Data::Tag(TAG_STRING_DATETIME, Box::new(Data::Text(v.to_raw())))))
+				}
+			}
+			Value::Uuid(v) => {
+				if SERIALIZATION_V2 {
+					Ok(Cbor(Data::Tag(TAG_UUID, Box::new(Data::Bytes(v.into_bytes().into())))))
+				} else {
+					Ok(Cbor(Data::Tag(TAG_STRING_UUID, Box::new(Data::Text(v.to_raw())))))
+				}
```

Comment:
CBOR is currently undocumented, so breaking it is not an issue. Let's drop support for string based UUIDs.

```suggestion
				Ok(Cbor(Data::Tag(TAG_UUID, Box::new(Data::Bytes(v.into_bytes().into())))))
```